### ISSUE TITLE
Design tools exploration: Contextual color controls and new panels for Background, and Elements

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#62752]
Extracted from discussion in https://github.com/WordPress/gutenberg/pull/60100#issuecomment-2180553281 and https://github.com/WordPress/gutenberg/pull/60151.

In container blocks like Group, the Color panel provides options to apply colors to child elements such as buttons, links, and headings.

<img width="937" alt="color" src="https://github.com/WordPress/gutenberg/assets/846565/a55293a6-3784-4fe5-9c15-52f9ede411e8">

While this feature is useful, it has limitations – for example, it doesn't allow for adjusting the _dimensions_ of buttons within the Group.

Another issue is the disorganized background options: the background color control is in the Color panel, while the background image control is in a separate Background Image panel.

### Contextual color controls
<img width="937" alt="panels" src="https://github.com/WordPress/gutenberg/assets/846565/8ee16faf-b0c9-4d29-9ea5-3b8cb25bee37">

I suggest moving text and background controls to more appropriate panels. The text color control could be relocated to the Typography panel, and the background color control to a new Background panel, alongside the background image control.

A dedicated Background panel would allow users to add multiple background images and reorder them as needed.

### Child elements

<img width="937" alt="elements" src="https://github.com/WordPress/gutenberg/assets/846565/1fa303fd-d0f4-480b-a3a3-5640cf6ae2e3">

To enhance functionality, a new 'Elements' panel could enable users to fully style child elements within a container. This approach is similar to styling blocks in Global styles, which has been quite effective.

---

Additionally, this reorganization could simplify the 'Overlay' option in the Cover block, combining background images and transparent background colors into a single control that can be applied to any block, not just the Cover block.

<img width="937" alt="overlay" src="https://github.com/WordPress/gutenberg/assets/846565/cfec1cac-52d2-4e91-ac74-c4c379d3e7dd">

cc @WordPress/gutenberg-design for thoughts.

**Comment by richtabor** [comment]
Interesting exploration. Would background controls open via a popover?

**Comment by jameskoster** [comment]
Precisely. There's a PR for that over here: https://github.com/WordPress/gutenberg/pull/60151
